packages/testlab/src/test-sandbox.ts

```diff
@@ -58,6 +58,16 @@ export class TestSandbox {
    */
   async reset(): Promise<void> {
     this.validateInst();
+
+    // Decache files from require's cache so future tests aren't affected incase
+    // a file is recreated in sandbox with the same file name but different
+    // contents after resetting the sandbox.
+    for (const key in require.cache) {
+      if (key.startsWith(this.path)) {
```

Comment:
Should we check that the `key` starts with `this.path + '/'` to ensure we don't accidentally decache directory names like `sandbox-123/file.js` or `sandbox.bak/file.js`? (Plus handle `\` on Windows.)

Not a big deal, maybe fixing this edge case is not worth the implementation complexity.